lxd/instance.go

```diff
@@ -174,6 +174,137 @@ func instanceCreateFromImage(s *state.State, r *http.Request, img *api.Image, ar
 	return inst, nil
 }
 
+func instanceRebuildFromImage(s *state.State, r *http.Request, inst instance.Instance, img *api.Image, op *operations.Operation) error {
+	// Validate the type of the image matches the type of the instance.
+	imgType, err := instancetype.New(img.Type)
```

Comment:
This section around image validation and transfer looks like its been duplicated?
In the interests of future maintenance, please can you extract the common logic and then call it from both old and new code paths please?